webapp/src/workspace.ts

```diff
@@ -340,27 +341,33 @@ export function installAsync(h0: InstallHeader, text: ScriptText) {
         .then(() => h)
 }
 
-export function duplicateAsync(h: Header, text: ScriptText): Promise<Header> {
+export function duplicateAsync(h: Header, text: ScriptText, rename?: boolean): Promise<Header> {
     let e = lookup(h.id)
     U.assert(e.header === h)
     let h2 = U.flatClone(h)
     e.header = h2
 
     h.id = U.guidGen()
-    let names = U.toDictionary(allScripts, e => e.header.name)
-    let n = 2
-    while (names.hasOwnProperty(h.name + " #" + n))
-        n++
-    h.name += " #" + n
-    let cfg = JSON.parse(text[pxt.CONFIG_NAME]) as pxt.PackageConfig
-    cfg.name = h.name
-    text[pxt.CONFIG_NAME] = JSON.stringify(cfg, null, 4)
+    if (rename) {
```

Comment:
If rename is false and there is a collision of name?

Reply:
id is refreshed, name collisions are allowed.

Reply:
hence how you can have so many Untitled projects.